primary/tests/integration_tests.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use config::{Parameters, WorkerId};
+use crypto::traits::Signer;
+use crypto::Hash;
+use crypto::{ed25519::Ed25519PublicKey, traits::KeyPair};
+use primary::CHANNEL_CAPACITY;
+use primary::{PayloadToken, Primary};
+use std::time::Duration;
+use store::{rocks, Store};
+use test_utils::{
+    certificate, committee_with_base_port, fixture_batch_with_transactions, fixture_header_builder,
+    keys, temp_dir,
+};
+use tokio::sync::mpsc::channel;
+use types::{
+    BatchDigest, Certificate, CertificateDigest, CertificateDigestProto, CollectionRetrievalResult,
+    GetCollectionsRequest, Header, HeaderDigest, ValidatorClient,
+};
+use worker::{SerializedBatchMessage, Worker, WorkerMessage};
+
+#[tokio::test]
+async fn test_get_collections() {
+    let keypair = keys().pop().unwrap();
+    let name = keypair.public().clone();
+    let signer = keypair;
+    let committee = committee_with_base_port(11_000);
+    let parameters = Parameters {
+        batch_size: 200, // Two transactions.
+        ..Parameters::default()
+    };
+
+    // Create a new test header store.
+    let header_map = rocks::DBMap::<HeaderDigest, Header<Ed25519PublicKey>>::open(
+        temp_dir(),
+        None,
+        Some("headers"),
+    )
+    .unwrap();
+    let header_store = Store::new(header_map);
+
+    // Create a new test certificate store.
+    let certificate_map = rocks::DBMap::<CertificateDigest, Certificate<Ed25519PublicKey>>::open(
+        temp_dir(),
+        None,
+        Some("certificates"),
+    )
+    .unwrap();
+    let certificate_store = Store::new(certificate_map);
+
+    // Create a new test payload store.
+    let payload_map = rocks::DBMap::<(BatchDigest, WorkerId), PayloadToken>::open(
+        temp_dir(),
+        None,
+        Some("payloads"),
+    )
+    .unwrap();
+    let payload_store = Store::new(payload_map);
+
+    // Create a new test batch store.
+    let batch_map = rocks::DBMap::<BatchDigest, SerializedBatchMessage>::open(
+        temp_dir(),
+        None,
```

Comment:
I believe you can use instead the NodeStorage https://github.com/MystenLabs/narwhal/blob/fecbfc9385e5599aed504d2b63de4b8763d4b6a2/node/src/lib.rs#L38 to initialise all the required stores.